lib/Transform/XTenMinimizeLiveTensors.cpp

```diff
@@ -86,11 +88,43 @@ bool isConvAddChained(Operation *op) {
              xilinx::xten::Conv2dTensorAddLReLUGlobalAveragePoolOp>(op);
 }
 
+bool isInCoreChainSubgraph(Operation *op) {
+  return (op != nullptr) && op->hasAttr("Reason") &&
+         op->getAttr("Reason").cast<StringAttr>() == "InCoreChain";
```

Comment:
The documentation isn't great, but [getAttrOfType](https://mlir.llvm.org/doxygen/IR_2Operation_8h_source.html#l00419) returns a `nullptr` if either the attribute is not present or is of the wrong type. Using that might give more elegant code. (Here and in a number of other places in the code.)